exporter/azuremonitorexporter/trace_to_envelope.go

```diff
@@ -0,0 +1,660 @@
+	data.Name = sb.String()
+
+	// Set the .Data property to .Name which contain the full RPC method
+	data.Url = data.Name
+
+	// data.Source should be the net.peer.name if available otherwise fallback to net.peer.ip
+	if attrs.NetworkAttributes.NetPeerName != "" {
+		data.Source = attrs.NetworkAttributes.NetPeerName
+	} else if attrs.NetworkAttributes.NetPeerIP != "" {
+		data.Source = attrs.NetworkAttributes.NetPeerIP
+	}
+}
+
+// Maps RPC Client Span to AppInsights RemoteDependencyData
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/rpc.md
+func fillRemoteDependencyDataRPC(span pdata.Span, data *contracts.RemoteDependencyData) {
+	attrs := copyAndExtractRPCAttributes(span.Attributes(), data.Properties, data.Measurements)
+
+	var sb strings.Builder
+
+	// Set the .Data property to .Name which contain the full RPC method
+	data.Data = data.Name
+
+	data.Type = attrs.RPCSystem
+
+	// Favor name over IP for data.Target
+	if attrs.NetworkAttributes.NetPeerName != "" {
+		sb.WriteString(attrs.NetworkAttributes.NetPeerName)
+	} else if attrs.NetworkAttributes.NetPeerIP != "" {
+		sb.WriteString(attrs.NetworkAttributes.NetPeerIP)
+	}
+
+	if attrs.NetworkAttributes.NetPeerPort != 0 {
+		sb.WriteString(":")
+		sb.WriteString(strconv.FormatInt(attrs.NetworkAttributes.NetPeerPort, 10))
+	}
+
+	data.Target = sb.String()
+}
+
+// Maps Database Client Span to AppInsights RemoteDependencyData
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/database.md
+func fillRemoteDependencyDataDatabase(span pdata.Span, data *contracts.RemoteDependencyData) {
+	attrs := copyAndExtractDatabaseAttributes(span.Attributes(), data.Properties, data.Measurements)
+
+	data.Type = attrs.DBType
+	data.Data = attrs.DBStatement
+	data.Target = attrs.DBURL
+}
+
+// Maps Messaging Consumer/Server Span to AppInsights RequestData
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md
+func fillRequestDataMessaging(span pdata.Span, data *contracts.RequestData) {
+	attrs := copyAndExtractMessagingAttributes(span.Attributes(), data.Properties, data.Measurements)
+
+	// TODO Understand how to map attributes to RequestData fields
+	data.Source = attrs.MessagingURL
+}
+
+// Maps Messaging Producer/Client Span to AppInsights RequestData
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md
+func fillRemoteDependencyDataMessaging(span pdata.Span, data *contracts.RemoteDependencyData) {
+	attrs := copyAndExtractMessagingAttributes(span.Attributes(), data.Properties, data.Measurements)
+
+	// TODO Understand how to map attributes to RemoteDependencyData fields
+	data.Data = attrs.MessagingURL
+	data.Type = attrs.MessagingSystem
+	data.Target = attrs.MessagingDestination
```

Comment:
I have thought that we always need to look at `net.peer.*` as a fallback for all types.  How do you think?

Reply:
I agree with this also.